Silicon/NXP/iMXPlatformPkg/Drivers/SmbiosConfigDxe/SmbiosConfigDxe.c

```diff
@@ -0,0 +1,878 @@
+  CONST CHAR16                      *DevicePathText;
+  EFI_FILE_HANDLE                   File;
+  EFI_FILE_INFO                     *FileInfo;
+  UINTN                             FileInfoSize;
+  EFI_SIMPLE_FILE_SYSTEM_PROTOCOL   *Fs;
+  EFI_DEVICE_PATH_PROTOCOL          *IntermediateDevicePath;
+  EFI_HANDLE                        MediaHandle;
+  EFI_FILE_HANDLE                   RootVolume;
+  EFI_STATUS                        Status;
+ 
+  File = NULL;
+  RootVolume = NULL;
+  Buffer = NULL;
+  DevicePath = NULL;
+  MediaHandle = NULL;
+  Fs = NULL;
+
+  //
+  // Find Smbios.ini file starting with Pcd device path
+  //
+  DevicePathText = (CONST CHAR16 *) FixedPcdGetPtr (PcdSmbiosOverrideDevicePath);
+  if ((DevicePathText == NULL) || (*DevicePathText == L'\0')) {
+    Status = EFI_INVALID_PARAMETER;
+    DEBUG ((DEBUG_ERROR, "%a: PcdStorageMediaPartition is unspecified\n", __FUNCTION__));
+    goto Exit;
+  }
+
+  DevicePath = ConvertTextToDevicePath (DevicePathText);
+  if (!DevicePath) {
+    DEBUG ((DEBUG_ERROR, "%a: Could not convert DevicePathText\n", __FUNCTION__));
+    return FALSE;
+  }
+
+  IntermediateDevicePath = DevicePath;
+  Status = gBS->LocateDevicePath (
+    &gEfiSimpleFileSystemProtocolGuid,
+    &IntermediateDevicePath,
+    &MediaHandle);
+
+  if (Status == EFI_NOT_FOUND) {
+    DEBUG ((DEBUG_INFO,"%a: %s FAT partition is not ready yet\n", __FUNCTION__, DevicePathText));
+    goto Exit;
+  }
+
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "%a: gBS->LocateDevicePath() failed. (Status=%r)\n", __FUNCTION__, Status));
+    goto Exit;
+  }
+
+  Status = gBS->HandleProtocol (
+    MediaHandle,
+    &gEfiSimpleFileSystemProtocolGuid,
+    (VOID**)&Fs
+    );
+
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "%a: gBS->HandleProtocol() failed. (Status=%r)\n", __FUNCTION__, Status));
+    goto Exit;
+  }
+
+  Status = Fs->OpenVolume (Fs, &RootVolume);
+  if (EFI_ERROR(Status)) {
+    DEBUG ((DEBUG_ERROR, "%a: Fs->OpenVolume() failed. (Status=%r)\n", __FUNCTION__, Status));
+    goto Exit;
+  }
+
+  Status = RootVolume->Open (
+    RootVolume,
+    &File,
+    L"Smbios.ini",
+    EFI_FILE_MODE_READ,
+    0);
+
+  if (EFI_ERROR(Status)) {
+    DEBUG ((DEBUG_ERROR, "%a: RootVolume->Open() failed. (Status=%r)\n", __FUNCTION__, Status));
+    goto Exit;
+  }
+
+  //
+  // Read File Info
+  // 
+  FileInfoSize = sizeof (EFI_FILE_INFO) + 1024;
+  FileInfo = AllocateZeroPool (FileInfoSize);
+  if (FileInfo == NULL) {
+    DEBUG ((DEBUG_ERROR, "%a: FileInfo allocation failed\n", __FUNCTION__));
+    goto Exit;
+  }
+
+  Status = File->GetInfo (
+                  File,
+                  &gEfiFileInfoGuid,
+                  &FileInfoSize,
+                  FileInfo
+                  );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "%a: Failed to get File Info (Status=%r)\n", __FUNCTION__, Status));
+    goto Exit;
+  }
+
+  BufferSize = (UINTN) FileInfo->FileSize + sizeof (CHAR16);
```

Comment:
Bounds check the FileSize. Expected file sizes are <1K Bytes. So limit File size to 10x this.